### DISCUSSION TITLE
Integrate shoelace design components in lit ssr

### DISCUSSION THREAD

**Discussion by adil-94** [lit/lit#4765] [Q&A]
Hi Team,
I want to integrate shoelace design components in my lit project with SSR.
The shoelace button component to be rendered at server and hydrate at client.
Iam using Koa server and roll up for bundling.
When I try to import shoelace styles or components inside my kit component it breaks the UI.

Can you please help me to use shoelace components in my kit project?

**Comment by KonnorRogers** [comment]
Hi there. I'm one of the maintainers of Shoelace. SSR is on our Roadmap, I have it working in our future 3.0 release of Shoelace called "Web Awesome". I don't know if a backport to Shoelace will happen simply because it was quite a bit of effort.

Unfortunately, in Shoelace's current state there's not much the Lit team can do to help you.

You _could_ have a bunch of DOM shims on your server for things like `MutationObserver`, `IntersectionObserver`, `querySelector` and a number of other client APIs, but it would be brittle and you would probably still hit hydration errors.

TLDR: Lit SSR _generally_ isn't supported in Shoelace currently. But we are working on it.

**Reply by adil-94** [reply]
Hi @KonnorRogers 
Thanks for the update.
When can we expect shoelace 3.0 to be out any rough estimation on this?
Also it would be great if you can share GitHub link for shoelace 3.0

**Reply by KonnorRogers** [reply]
Unfortunately Shoelace 3.0 (renamed to Web Awesome) is still in private development right now so nothing I can link to.

Those who backed our [kickstarter](https://www.kickstarter.com/projects/fontawesome/web-awesome?ref=4gkt5y) did get a sneak peek at an alpha version without the source code.

Our expected release is April of 2025.
